docs/proposals/registry/registry-operator.md

```diff
@@ -0,0 +1,100 @@
+# Devfile Registry Operator Proposal
+
+## Problem
+Currently, the devfile registry is deployed onto a running cluster by `kubectl apply`-ing a few yaml files. This works fine for a proof of concept or doing a quick demo, but there’s a few problems with this approach that make it infeasible long-term:
+
+1) Configuration is difficult.
+   
+   For example, the current approach makes it difficult to programatically configure TLS and authentication for the OCI registry server when deploying.
+
+2) Deployment isn't user friendly and hard to tell how/why something went wrong
+
+3) Updates to the registry instance are not straightfoward
+
+    Requires the registry administrator to manually edit and update the devfile registry deployment resource.
+
+3) Deployment isn’t the same on Kubernetes and OpenShift.
+
+   To handle this, we currently have to maintain two separate sets of yaml files for deploying the registry on Kubernetes or OpenShift.
+
+## Solution
+Implementing an operator to programatically deploy devfile registries will streamline the deployment process on both Kubernetes and OpenShift, as well as provide greater flexibility for customization and configuration when deploying it (especially as we add support for deploying with TLS and authentication enabled). It will also provide us the ability to onboard the devfile registry to OperatorHub in the future, if we wish.
+
+## DevfileRegistry Custom Resource
+
+### API Group:
+`registry.devfile.io`
+
+### API Version
+`v1alpha1`
+
+### Resource Kind
+`DevfileRegistry`
+
+### Scope
+`Namespaced`
+
+### Fields
+
+`devfileIndexImage` (string) Image used to bootstrap the OCI registry with devfiles and host the index.json.
+
+`ociRegistryImage` (string) Image used for the OCI registry instance. Defaults to `registry:2.7.1`
+
+#### TLS
+`enabled` (bool) TLS mode for the devfile registry. Defaults to true. If on OpenShift, the cluster's routing certificate will be re-used. If on Kubernetes, cert-manager will be used to provision a TLS secret.
+
+`tlsSecretName` (string) Optional pre-existing Kubernetes secret for TLS termination on the ingress resource. If set, cert-manager will **not** be used to provision a TLS secret.
+
+#### Storage
+`enabled` (bool) Enables persistent storage for the registry. Defaults to true. Can be disabled for development or testing only.
```

Comment:
Sorry for being late for party )

From https://docs.google.com/document/d/1rQHCp4SWslWWJv5KK3A_iXHvgbqjsuDkSKDD72ifJio/edit#heading=h.cexfsm1d9e6n
> Update
The registry admin updates the devfile registry resource to pull down any new/updated stacks
The registry operator stands up a new registry deployment, that uses the updated container images with the updated devfile stacks
Once the new registry deployment is ready, the operator updates the registry ingress resource to point to the new registry, and tears down the old registry.

Does it mean that we require ReadWriteMany access mode?

Reply:
Nope, it'll be a new volume as well that's brought up as well, so still only ReadWriteOnce that's used.